lldb/source/Commands/CommandObjectTarget.cpp

```diff
@@ -1618,12 +1621,15 @@ static uint32_t LookupSymbolInModule(CommandInterpreter &interpreter,
         if (symbol->ValueIsAddress()) {
           DumpAddress(
               interpreter.GetExecutionContext().GetBestExecutionContextScope(),
-              symbol->GetAddressRef(), verbose, all_ranges, strm);
+              symbol->GetAddressRef(), verbose, all_ranges, strm,
+              use_color ? name : nullptr);
```

Comment:
I think there might be a bug here. We have this `name_is_regex` and this is not checking it.

Should it be:
```
use_color && name_is_regex ? name : nullptr
```

So that when we have colours enabled, but do not search by regex, we do not get coloring.

First step to confirm would be to add a test that checks this and see if it fails.

Reply:
Enable colours
symbol lookup name (no regex option)
Expect that the results have no colouring

Reply:
Ohh didn't realize this one. Thanks for pointing out. Have updated the code and added a test case using image lookup command without regex flag.

Reply:
Yeah, I'm not 100% it should be regex only but my logic was that non-regex search looks for identical names, so the highlighting doesn't help you understand the result any more than the plain text does:
```
(lldb) image lookup -s ma
(lldb) image lookup -s main
1 symbols match 'main' in /tmp/test.o:
        Address: test.o[0x000000000000074c] (test.o.PT_LOAD[0]..text + 316)
        Summary: test.o`main
```
The name will only ever show up if it matched exactly.

Only when you've got something that could match in different ways does the colour help me:
```
(lldb) image lookup -s ma --regex
3 symbols match the regular expression 'ma' in /tmp/test.o:
        Name: __libc_start_main@@GLIBC_2.17
        Value: 0x0000000000000000
        Address: test.o[0x000000000000074c] (test.o.PT_LOAD[0]..text + 316)
        Summary: test.o`main
        Address: test.o[0x00000000000005e0] (test.o.PT_LOAD[0]..plt + 48)
        Summary: test.o`symbol stub for: __libc_start_main
<...>
```
Because it tells me what it matched.

But I am bringing my bias in here, I understand what all the lines of output mean. Maybe if I didn't, I would appreciate knowing where my search term shows up?

On the other hand, highlighting the same thing 99 times might just be noise on top of an already large amount of output. So I'm being 1. biased and 2. conservative.

Perhaps you would find the highlighting useful in both modes, if so we can keep it.